__tests__/app.test.js

```diff
@@ -120,3 +120,40 @@ describe('GET /api/topics', () => {
     
   });
 
+  describe("GET /api/articles/:article_id/comments", () => {
+    test("200 OK: all comments for an article", () => {
+      return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then(({ body }) => {
+          const { comments } = body;
+          expect(comments).toBeInstanceOf(Array);
+          comments.forEach((comment) => {
+            expect(comment).toHaveProperty("comment_id");
+            expect(comment).toHaveProperty("votes");
+            expect(comment).toHaveProperty("created_at");
+            expect(comment).toHaveProperty("author");
+            expect(comment).toHaveProperty("body");
+            expect(comment).toHaveProperty("article_id");
+          });
+        });
+    });
+  
+    test("404 Not Found: valid api but no ID", () => {
+      return request(app)
+        .get("/api/articles/99999/comments")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Not Found");
+        });
+    });
+  
+    test("400 Bad Request: error message when passed invalid article ID", () => {
+      return request(app)
+        .get("/api/articles/noID/comments")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad Request");
+        });
+    });
```

Comment:
Nice work - I think we could consider one more scenario:

What happens if the article id is valid and it exists in the database, but there just so happens to be no comments associated with it. This shouldn't be an error, we should return an empty array.

Hint: we will need to check if the article id exists first